server/build/babel/preset.js

```diff
@@ -7,10 +7,21 @@ const productionPlugins = isProduction ? [
   require.resolve('babel-plugin-transform-react-remove-prop-types')
 ] : []
 
+const es2015Config = {}
+
+// Do not transform ES2015 modules if we are inside Next.js
+// That's because webpack2 knows how to handle it.
+// (And that's how it can do code splitting)
+//
+// But in other environements like Jest, we should transform it.
+if (process.env.INSIDE_NEXT) {
```

Comment:
Then the problem when used with Jest. Both Jest and Next.js reads from the same config.
So, we have to add different env and provide different babel setups.

If we go this approach, there's no need to take this PR at all.
We can simply ask them to use `es2015` + `next/babel`